ui/components/app/terms-of-use-popup/terms-of-use-popup.js

```diff
@@ -0,0 +1,1155 @@
+import React, { useContext, useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import { I18nContext } from '../../../contexts/i18n';
+import Button from '../../ui/button';
+import Popover from '../../ui/popover';
+import {
+  AlignItems,
+  FLEX_DIRECTION,
+  TextVariant,
+} from '../../../helpers/constants/design-system';
+import { Text } from '../../component-library';
+import Box from '../../ui/box';
+import CheckBox from '../../ui/check-box/check-box.component';
+
+export default function TermsOfUsePopup({ onAccept }) {
+  const t = useContext(I18nContext);
+  const [isTermsOfUseChecked, setIsTermsOfUseChecked] = useState(false);
+  const [shouldShowScrollButton, setShouldShowScrollButton] = useState(true);
+
+  const popoverRef = useRef();
+  const bottomRef = React.createRef();
+
+  const handleScrollDownClick = (e) => {
+    e.stopPropagation();
+    bottomRef.current.scrollIntoView({
+      behavior: 'smooth',
+    });
+    setShouldShowScrollButton(false);
+  };
+
+  return (
+    <Popover
+      className="terms-of-use__popover"
+      popoverRef={popoverRef}
+      title={t('termsOfUseTitle')}
+      showScrollDown={shouldShowScrollButton}
+      onScrollDownButtonClick={handleScrollDownClick}
+      footerProps={{
+        justifyContent: AlignItems.center,
+        flexDirection: FLEX_DIRECTION.COLUMN,
+      }}
+      footer={
+        <>
+          <Button
+            type="primary"
+            className="terms-of-use__button"
+            onClick={onAccept}
+            disabled={!isTermsOfUseChecked}
+            data-testid="terms-of-use-accept-button"
+          >
+            {t('accept')}
+          </Button>
+          <Text as="p" marginTop={4} className="terms-of-use__footer-text">
+            Please scroll to read all sections
+          </Text>
+        </>
+      }
+    >
+      <div className="terms-of-use">
```

Comment:
We can use Box here instead of div
```suggestion
      <Box className="terms-of-use">
```